### PR TITLE
Add support for Slovenian holidays

### PR DESCRIPTION
**Pull request by juref**
Added Slovenian holidays

### CODE REVIEW

**Comment by juref** [comment]
Maybe we could add a new method beside `getName('Y-m-d')` `getTranslatedName('Y-m-d')` to return the translated name. I see two good options:

`Holidays::for('be')->getName('2024-01-01'); // Nieuwjaar` - original
`Holidays::for('be')->getTranslatedName('2024-01-01'); // New year` - option 1
`Holidays::for('be')->getName('2024-01-01')->translate('en'); // New year` - option 2

In both options, we would need to add a translation file or reference array.

File: src/Countries/Slovenia.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        return 'si';
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function allHolidays(int $year): array
+    {
+        return array_merge([
+            'Novo leto' => '01-01', // New Year's Day
+            'Novo leto 2' => '02-01', // New Year's Day, yes it's a second day
```
Comment (nexxai) [inline src/Countries/Slovenia.php:19]: This should be `01-02` if the day is January 2nd, right?

File: tests/.pest/snapshots/Countries/SloveniaTest/it_can_calculate_slovenian_holidays.snap [hunk 2]
```diff
@@ -0,0 +1,54 @@
+[
+    {
+        "name": "Novo leto",
+        "date": "2024-01-01"
+    },
+    {
+        "name": "Novo leto 2",
+        "date": "2024-02-01"
```
Comment (nexxai) [inline tests/.pest/snapshots/Countries/SloveniaTest/it_can_calculate_slovenian_holidays.snap:8]: Same here

**Review by nexxai — COMMENTED** [review]
I believe the formatting for these dates should be "MM-DD" to match all of the others

File: src/Countries/Slovenia.php (hunk 1 above)
Comment (juref) [reply src/Countries/Slovenia.php]: Thank you for noticing. In out country the default is  DD-MM that's why I accidentally put it this way.

**Comment by Nielsvanpach** [comment]
> Maybe we could add a new method beside `getName('Y-m-d')` `getTranslatedName('Y-m-d')` to return the translated name. I see two good options:


I'm open to accept a good PR which implements translations. https://github.com/spatie/holidays/discussions/131

File: src/Countries/Slovenia.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+use Carbon\CarbonImmutable;
+
+class Slovenia extends Country
+{
+    public function countryCode(): string
+    {
+        return 'si';
+    }
+
+    /** @return array<string, CarbonImmutable> */
```
Comment (Nielsvanpach) [inline src/Countries/Slovenia.php:14]: can be removed

File: src/Countries/Slovenia.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            'Dan spomina na mrtve' => '11-01', // Remembrance Day
+            'Božič' => '12-25', // Christmas Day
+            'Dan samostojnosti in enotnosti' => '12-26', // Independence and Unity Day
+        ], $this->variableHolidays($year));
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        $easter = CarbonImmutable::createFromTimestamp(easter_date($year))
```
Comment (Nielsvanpach) [inline src/Countries/Slovenia.php:37]: use `$this->easter($year)` instead. Only available if your branch is rebased on the repo's main branch.

**Comment by Nielsvanpach** [comment]
Something seem to went wrong with rebasing. Could you make a clean PR with only commits related to Slovenia? Thanks!

**Comment by juref** [comment]
Sure, no problem.

**Comment by juref** [comment]
New PR link:
https://github.com/spatie/holidays/pull/139

**Pull request closed without merging**